### PR TITLE
Challenge 1 implementation

### PR DESCRIPTION
**Pull request by nicoletaserban-cz**

### CODE REVIEW

**Review by teodora-bobirneci — COMMENTED** [review]
Did you forget to commit your docker compose file? And also the localsetup initialization script.

Later edit: nvm, found them 😅

File: letter-sender/src/main/java/com/christmas/letter/sender/model/Letter.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class Letter {
+    String email;
```
Comment (teodora-bobirneci) [inline letter-sender/src/main/java/com/christmas/letter/sender/model/Letter.java]: I would expect to see some validations here, the email should be required, it should match text@text.domain. You could use @Data instead of @Getter, @Setter etc, and the properties should be private.

File: letter-sender/src/test/java/LetterControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+@AutoConfigureMockMvc
+@ExtendWith(OutputCaptureExtension.class)
+@SpringBootTest(classes = LetterSenderApplication.class)
+public class LetterControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Container
+    static LocalStackContainer localStackContainer;
```
Comment (teodora-bobirneci) [inline letter-sender/src/test/java/LetterControllerIntegrationTest.java]: Let's extract this code that does the testcontainers initialization in some sort of base class and keep the test class clean.

File: letter-sender/src/test/java/LetterControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        registry.add("spring.cloud.aws.sns.region", localStackContainer::getRegion);
+        registry.add("spring.cloud.aws.sns.endpoint", localStackContainer::getEndpoint);
+        registry.add("aws.sns.topic.arn", () -> TOPIC_ARN);
+
+        registry.add("spring.cloud.aws.sqs.region", localStackContainer::getRegion);
+        registry.add("spring.cloud.aws.sqs.endpoint", localStackContainer::getEndpoint);
+    }
+
+    @Test
```
Comment (teodora-bobirneci) [inline letter-sender/src/test/java/LetterControllerIntegrationTest.java:47]: Let's also add some tests for all negative scenarios, letter body is missing, email is missing, email is invalid, etc. You can use @ParameterizedTest
https://www.baeldung.com/parameterized-tests-junit-5

**Review by tudorcodrea — COMMENTED** [review]
pretty straightforward send workflow to SNS. clean and clear
Usually in production we are having DTOs coming in from REST client and domain objects that separate the domain from DTO model. but in this case the mapping is 1-1 so looks good. Don't forget to add validation upon DTO as Teodora said and negative testing

File: letter-sender/src/main/java/com/christmas/letter/sender/model/Letter.java (hunk 1 above)
Comment (nicoletaserban-cz) [reply letter-sender/src/main/java/com/christmas/letter/sender/model/Letter.java]: Done

File: letter-sender/src/test/java/LetterControllerIntegrationTest.java (hunk 2 above)
Comment (nicoletaserban-cz) [reply letter-sender/src/test/java/LetterControllerIntegrationTest.java]: done

File: letter-sender/src/test/java/LetterControllerIntegrationTest.java (hunk 3 above)
Comment (nicoletaserban-cz) [reply letter-sender/src/test/java/LetterControllerIntegrationTest.java:47]: done

**Review by tudorcodrea — APPROVED** [review]

File: letter-sender/src/test/java/LetterControllerIntegrationTest.java (hunk 3 above)
Comment (teodora-bobirneci) [reply letter-sender/src/test/java/LetterControllerIntegrationTest.java:47]: I would also validate the error message, for example "email is required" or whatever details we provide for the user.

**Review by teodora-bobirneci — APPROVED** [review]

**Review by teodora-bobirneci — APPROVED** [review]
